### PR TITLE
Bugfixes.

### PR DESCRIPTION
**Pull request by deathbaba**

### CODE REVIEW

File: src/gzip_wrapper.h [hunk 1]
```diff
@@ -21,6 +21,7 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 *******************************************************************************/
+#include <cstring> // std::memset
```
Comment (dkorolev) [inline src/gzip_wrapper.h]: Two spaces before `//`.

File: src/gzip_wrapper.h [hunk 2]
```diff
@@ -41,18 +42,10 @@ struct GzipErrorException : public std::exception {
...
-  }
-};
-
-inline std::string Gzip(const std::string& data_to_compress) throw(GzipErrorException) {
-  z_stream z = {};
+// Throws GzipErrorException on any gzip processing error.
+inline std::string Gzip(const std::string& data_to_compress) {
+  z_stream z;
+  std::memset(&z, 0, sizeof(z));
   int res = ::deflateInit2(&z, Z_BEST_COMPRESSION, Z_DEFLATED, 15 + 16, 8, Z_DEFAULT_STRATEGY);
```
Comment (dkorolev) [inline src/gzip_wrapper.h]: const int res

File: src/gzip_wrapper.h [hunk 3]
```diff
@@ -77,8 +70,20 @@ inline std::string Gzip(const std::string& data_to_compress) throw(GzipErrorExce
   throw GzipErrorException(res, z.msg);
 }
 
-inline std::string Gunzip(const std::string& data_to_decompress) throw(GunzipErrorException) {
-  z_stream z = {};
+struct GunzipErrorException : public std::exception {
+  std::string msg_;
```
Comment (dkorolev) [inline src/gzip_wrapper.h]: Why not `const std::string msg_;` and initialization?

**Comment by dkorolev** [comment]
LGTM++, and I'll merge it.

Side note: I believe `z_stream z = {0};` instead of `z_stream z = {};` does the nullification, but would too prefer `::memset()` when using C code myself.

Thanks,
Dima

**Comment by deathbaba** [comment]
I have checked it, gcc still warns about not initialized members (clang does it correctly).

**Comment by dkorolev** [comment]
Good to know, thanks for checking!

**Pull request merged**